### PR TITLE
Fix jsx-curly-spacing: spacing around spread operator

### PR DESCRIPTION
**Pull request by gitim**
Fixes https://github.com/yannickcr/eslint-plugin-react/issues/606
Added two type of tests:
* creating jsx elements using only JSXSpreadAttribute syntax
* creating jsx elements using both JSXSAttribute  and JSXSpreadAttribute syntax

### CODE REVIEW

File: lib/rules/jsx-curly-spacing.js [hunk 1]
```diff
@@ -143,13 +143,14 @@ module.exports = function(context) {
...
-   * @param {Token} penultimate The penultimate token to check (should be last before closing brace)
-   * @param {Token} last The last token to check (should be closing brace)
    * @returns {void}
    */
-  function validateBraceSpacing(node, first, second, penultimate, last) {
+  function validateBraceSpacing(node) {
+    var first = context.getFirstToken(node);
+    var last = sourceCode.getLastToken(node);
+    var second = context.getTokenAfter(first);
+    var penultimate = sourceCode.getTokenBefore(last);
```
Comment (gitim) [inline lib/rules/jsx-curly-spacing.js]: just put these declarations inside the function to reduce copy pasting

File: tests/lib/rules/jsx-curly-spacing.js [hunk 2]
```diff
@@ -83,6 +83,88 @@ ruleTester.run('jsx-curly-spacing', rule, {
...
+      '...bar',
+      '}/>;'
+    ].join('\n'),
+    options: ['always'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App foo={bar/* comment */} {...baz/* comment */} />;',
+    options: ['never'],
+    parserOptions: parserOptions,
+    parser: 'babel-eslint'
```
Comment (ljharb) [inline tests/lib/rules/jsx-curly-spacing.js]: why do you need the babel-eslint parser for jsx object spread? that should only be needed for JS object spread.

Comment (gitim) [inline tests/lib/rules/jsx-curly-spacing.js]: ah, sorry, just copy pasted and modified this case https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/jsx-curly-spacing.js#L81-L86

Comment (ljharb) [inline tests/lib/rules/jsx-curly-spacing.js]: hm, i wouldn't expect that one to need it either

Comment (gitim) [inline tests/lib/rules/jsx-curly-spacing.js]: this test case was added in https://github.com/yannickcr/eslint-plugin-react/pull/584, and it seems that the issue wasn't related with babel-eslint parse, so I think it is safe to remove them both

**Comment by yannickcr** [comment]
Merged, thanks!

**Pull request closed without merging**